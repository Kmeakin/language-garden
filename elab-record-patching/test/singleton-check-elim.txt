let _ : fun (A : Type) (x : A) (sing-x : A [= x]) -> A :=
  fun A x sing-x := sing-x;
  --                ^^^^^^ Elaborates to `#sing-elim sing-x x`
  --                       when checking `sing-x` against `A`

Type
