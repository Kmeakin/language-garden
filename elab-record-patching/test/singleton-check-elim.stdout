<input> : Type :=
  let _ : fun (A : Type) (x : A) (sing-x : A [= x]) -> A :=
    fun A x sing-x := #sing-elim sing-x x;
  Type
